keps/sig-architecture/3138-increase-reliability-bar/README.md

```diff
@@ -0,0 +1,415 @@
+- test coverage (the details of how to measure it are to-be-decided)
+- [ideal but unlikely due to inability to get the data] number of outages
+  reported by Kubernetes providers
+
+
+## Design Details
+
+As mentioned above, we propose to divide the effort into 4 milestones.
+Below we describe these milestones, focusing on the first milestone that
+will be executed first as a pilot.
+
+### [Milestone 1] Existing tests failures/flakiness
+
+When a test fails (especially the end-to-end test), the reason for the failure
+is often unclear. All of the following reasons are theoretically possible:
+- test-related issue (e.g. test written in non-deterministic way, incorrect
+  CI job configuration, etc.)
+- a bug in the exercised component
+- a bug in another component (potentially owned by a different SIG)
+- some underlying infrastructure-related issues (e.g. cloud-provider outage)
+
+For sustainability, we should figure out a path towards understanding them.
+Obviously we’re extremely far from being able to do that now, but without any
+work the amount of necessary work to get there can only increase - it will
+not decrease on its own.
+
+To make the effort trackable and sustainable we propose the following:
+1. We will limit the effort only to the test jobs that are part of
+   “master release blocking” and “master release informing” tabs.
+   - If really needed we may consider extending the set of jobs in scope,
+     but this is not a plan of record as of now.
+
+1. We will use [Sippy][] as our source of truth for determining the health
+   of the individual jobs and tests.
+   We’ve already set up [Kubernetes Sippy][] that is monitoring the jobs
+   described above.
+   We will focus on the “Top failing tests” part of it.
+
+1. People may use https://storage.googleapis.com/k8s-triage/index.html for
+   finding instances of the failing tests that they can use for debugging purposes.
+
+1. We will be periodically looking at the “Top failing parts” to:
+   - Open issues (labeled with “kind/reliability-issue” and appropriate SIG)
+     for the worst one such that:
+     - There should be at most 3 open issues for a given SIG, with the caveat
+       of significant regressions - if the test pass rate goes down by more
+       than 10% week over week, that justifies opening additional ones
+     - We considered automatic issue filing but decided not to start with it due to:
+       - CI signal experience with flakes (e.g. system-level issues, handling
+         duplicates etc.)
+       - people tending to ignore bot-originating issues
+     - The proper labeling of issues would mean that people would be able to easily
+       find issues currently opened for a given SIG, via a simple github query like
+       this `https://github.com/kubernetes/kubernetes/issues?q=is%3Aopen+is%3Aissue+label%3Asig%2Ffoo+label%3Akind%2Freliability-issue`
+   - Notify the SIG on Slack asking for explicit acknowledgement on the issue
```

Comment:
I foresee this to be a point of conflict or for bikeshedding, 
what if the notification should use multiple channels in parallel, slack and email to start with?

Reply:
You mean saying here "Notify the SIG on both Slack and email (and whatever else you have on your mind) ..."?

What would it give us? Given that we need explicit ack from them, what multiple channels give us?

Reply:
yeah, forget that comment, I don't have good answer sorry, just pick one channel seems the best solution,